File: GFG/Arrays/maxconsone.py
def maxone(arr):
    n=len(arr)
    i,j=0,0
    maxq=0
    while (i<n and j<n):
        if(arr[i]==0):
            i+=1
            j+=1
        elif(arr[j]==1):
            j+=1
        elif (arr[j]==0):
            maxq=max(maxq,j-i)
            i=j
    return maxq

brr=[1,1,1,0,0,0,0,1,1,1,1,0,0,0,1,1]    
# brr=[0,0,0,0,0,0,0,0,0,0,0]
print(maxone(brr))

File: GFG/Arrays/maxconsone.cpp
#include<iostream>
#include<vector>
using namespace std;

int maxone(vector<int> &arr){
    int n = arr.size();
    int i=0,j=0;
    int maxq=0;
    while(i<n && j<n){
        if (arr[i]==0){i++;j++;}
        else if(arr[j]==1){j++;}
        else if (arr[j]==0){
            maxq=max(maxq,j-i);
            i=j;
        }
    }
    return maxq;
}
int main (){
    //vector<int> arr = {0,0,0,0,0,1,0,0,0,0,0,0};
    vector<int> arr = {1,1,1,0,0,0,0,1,1,1,1,0,0,0,1,1};
    cout<<maxone(arr);
    return 0;
}